Marlin/Configuration.h

```diff
@@ -1630,6 +1630,24 @@
 
 #define Z_PROBE_LOW_POINT          -2 // Farthest distance below the trigger-point to go before stopping
 
+// For M851 give a range for adjusting the X probe offset
+#ifndef Z_PROBE_OFFSET_XRANGE_MIN
+  #define Z_PROBE_OFFSET_XRANGE_MIN -50
+#endif
+
+#ifndef Z_PROBE_OFFSET_XRANGE_MAX
+  #define Z_PROBE_OFFSET_XRANGE_MAX 50
+#endif
+
+// For M851 give a range for adjusting the Y probe offset
+#ifndef Z_PROBE_OFFSET_YRANGE_MIN
+  #define Z_PROBE_OFFSET_YRANGE_MIN -50
+#endif
+
+#ifndef Z_PROBE_OFFSET_YRANGE_MAX
+  #define Z_PROBE_OFFSET_YRANGE_MAX 50
+#endif
```

Comment:
An issue with such an approach is that `Z_PROBE_OFFSET_YRANGE_MAX` would be defined even if the probe type is not selected.

I would like to wrap it with something like `#if HAS_PROBE_XY_OFFSET ...`, so probe-related settings are undefined in case the probe type itself is not declared.

At the same time, SanityCheck could produce an error if `Z_PROBE_OFFSET_YRANGE...` is declared but `HAS_PROBE_XY_OFFSET` is not.


I think "simple" bits like `HAS_BED_PROBE`, `HAS_PROBE_XY_OFFSET`, `PROBE_TRIGGERED_WHEN_STOWED_TEST` could be factored to `Marlin/src/inc/BedProbeType.h`, so that `BedProbeType.h` should be included right after `probe type` declaration in `Configuration.h`. Then `Configuration.h` could be written in such a way that it does not even declare properties like `NOZZLE_TO_PROBE_OFFSET` if probe type is void. I guess it could reduce the risks of accidental use of unrelated properties.

WDYT?

Reply:
This is not a problem. Not existent, not potential - nothing. Please stop spamming.

Reply:
Let's see where we end up after a bit of review and cleanup.